Oppdatert Modal Context til nytt `createContext`-api

KenAJoh opened the pull request:
For at `createContext` skal kunne fungere utenfor Provider-scope, er `strict` lagt til som en option i `createContext`

KenAJoh committed:
:recycle: Oppdatert hvordan ModalContext leveres

KenAJoh committed:
:label: onBeforeClose krever nå boolean

KenAJoh committed:
:recycle: Undo rename av Modal.types for å unngå conflict

KenAJoh committed:
:memo: Changeset

HalvorHaugan commented on @navikt/core/react/src/modal/modal.stories.tsx:
```diff
@@ -145,7 +145,7 @@ export const WithUseState: StoryFn = () => {
 
 export const EmptyHeader: StoryFn = () => (
-  <Modal open>
+  <Modal open aria-label="123">
```
Var det meningen å ta med dette?

KenAJoh replied:
Fjernet 👍 Hang igjen etter litt testing av typer

HalvorHaugan commented on @navikt/core/react/src/tooltip/Tooltip.tsx:
```diff
@@ -121,7 +115,7 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
     const arrowRef = useRef<HTMLDivElement | null>(null);
-    const modalContext = useContext(ModalContext);
+    const modalContext = useModalContext();
     const rootElement = modalContext ? modalContext.ref.current : undefined;
```
Modal med Tooltip inni funker ikke nå. Dette ble fanget opp av Chromatic, men synd at det ikke vises noe sted i GitHub.
https://www.chromatic.com/build?appId=5f801fb2aea7820022de2936&number=5099

KenAJoh replied:
Nice catch, var siden React by-default sletter `ref` når komponenten ikke er forwardRef. Oppdaterte Provider til å være forwardRef nå som fikset feilen.

Enig i at det har vært fint å få en heads-up i github. Burde få satt opp Storybook test-runner https://storybook.js.org/docs/writing-tests/test-runner

HalvorHaugan commented on @navikt/core/react/src/util/create-context.tsx:
Har noen ideer til forbedringer av createContext som jeg ønsker å diskutere.

KenAJoh replied:
🙌 🙌 
Er bare å starte en huddle i utviklerkanalen/melding når det passer

KenAJoh committed:
:fire: Fjernet unødvendig story-oppdatering

KenAJoh committed:
:bug: Fikset forwardRef i createContext

I tilfellene der ref blir sendt inn som en context-value, ble den tidligere slettet av React (native oppførsel)
ForwardRef lar oss beholde ref og sende den inn som context-value

HalvorHaugan committed:
Endring av strict

HalvorHaugan committed:
null -> undefined

KenAJoh committed:
Merge branch 'main' into v6-modal-types-v
